From bf6c26bf020f5a39b52c4cc718cf7a9dfbc165a8 Mon Sep 17 00:00:00 2001
From: chewitt <github@chrishewitt.net>
Date: Mon, 12 Aug 2019 02:55:09 +0000
Subject: [PATCH 152/182] WIP: arm64: dts: meson-g12b-odroid-n2: add battery
 rtc support

Changes suggested in Odroid forum [0] to enable battery RTC support

[0] https://forum.odroid.com/viewtopic.php?f=176&t=33993&start=300#p262877

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../boot/dts/amlogic/meson-g12b-odroid-n2.dts      | 41 ++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
index 9d98502..2f9247b 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
@@ -458,6 +458,47 @@
 	linux,rc-map-name = "rc-odroid";
 };
 
+&i2c3_sda_a_pins {
+	mux {
+		groups = "i2c3_sda_a";
+		function = "i2c3";
+		bias-pull-up;
+		drive-strength-microamp = <3000>;
+	};
+};
+
+&i2c3_sck_a_pins {
+	mux {
+		groups = "i2c3_sck_a";
+		function = "i2c3";
+		bias-pull-up;
+		drive-strength-microamp = <3000>;
+	};
+};
+
+&i2c3 {
+	status = "okay";
+	pinctrl-names = "default","gpio_periphs";
+	pinctrl-0 = <&i2c3_sck_a_pins &i2c3_sda_a_pins>;
+	pinctrl-1 = <&i2c3_to_gpioa>;
+	clock-frequency = <100000>; /* default 100k */
+
+	pcf8563: rtc@51 { /*I2C-bus slave address: read A3h and write A2h*/
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+};
+
+&periphs_pinctrl {
+	i2c3_to_gpioa:i2c3_gpioa {
+		mux {
+			groups = "GPIOA_14", "GPIOA_15";
+			function = "gpio_periphs";
+			drive-strength-microamp = <3000>;
+		};
+	};
+};
+
 &pwm_ab {
 	pinctrl-0 = <&pwm_a_e_pins>;
 	pinctrl-names = "default";
-- 
2.7.4

